packages/react/src/input/input.react.tsx

```diff
@@ -1,17 +1,19 @@
 import { c, classy, InputProps as BaseProps, m } from '@onfido/castor';
-import React from 'react';
+import React, { forwardRef } from 'react';
 
-export const Input = ({
-  type = 'text',
-  invalid,
-  className,
-  ...restProps
-}: InputProps): JSX.Element => (
-  <input
-    {...restProps}
-    type={type}
-    className={classy(c`input`, m({ invalid }), className)}
-  />
-);
+export const Input = forwardRef(_Input) as typeof _Input;
```

Comment:
I'm really unsure. I have an actual use case for Search in specific, which is

```ts
useEffect(() => {
    if (!search) return;
    // focus and move cursor to end
    searchRef.current?.focus();
    searchRef.current?.setSelectionRange(-1, -1);
  }, [search]);
```

But it could be implemented in a different way, e.g.

```ts
export function Input({ focus }) {
  const ref = useRef<HTMLInputElement>();
  
  useEffect(() => {
    if (!focus) return;
    // focus and move cursor to end
    ref.current?.focus();
    ref.current?.setSelectionRange(-1, -1);
  }, [focus]);
  
  return <input ref={ref} />
}
```

I prefer limited APIs, but it's hard to preemptively cover all use cases, we'll probably need to keep tinkering.
Like, what if `blur` is now needed?

Reply:
As mentioned some time earlier, I'd be pro having such by default on components.

Reply:
Other than the horrific type juggling `const Foo = forwardRef(_Foo) as typeof _Foo;`, there's also this:

![image](https://user-images.githubusercontent.com/18623773/102387208-01bb6700-3fc8-11eb-92f0-ab1571ba1db0.png)

Which probably means we'll need `_Foo.displayName = 'Foo';` 😞

Reply:
If you can find a nice solution for both problems, I'm happy to expose `ref`s for all base components (1:1 to an HTML element).

Reply:
I've create a helper `withRef` that "solves" both problems, please take a look and let me know what you think.

Reply:
What's the plan, to forward all components with separate PR, or not really? Case by case?

Reply:
I think it makes sense to forward all form-related/input components, we can do that on a separate PR, yea.

For others I think case by case is better, like `Icon` probably doesn't need a ref?

Reply:
Not sure, we can start with form components yes, then see if there's a need for others. Thanks!